source/developers/frontend_translation.markdown

```diff
@@ -0,0 +1,59 @@
+---
+layout: page
+title: "Translate Home Assistant frontend"
+description: "How to translate the frontend for Home Assistant."
+date: 2017-10-27 01:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+ha_release: 0.57
+---
+
+This will guide you how to translate the Home Assistant frontend to your language. Currently there are only a few translatable phrases - more will be added soon.
+Translations are defined in the [translations](https://github.com/home-assistant/home-assistant-polymer/tree/master/translations) folder of [home-assistant-polymer repository](https://github.com/home-assistant/home-assistant-polymer) at GitHub. The files are JSON formatted and named after their language tags.
+
+Example:
+```
+translations/en.json
+translations/en-US.json
+translations/de.json
+```
+
+Language subtags will inherit from their parent, so `en-US.json` should only contain the entries that will be different from `en.json`. All language tags inherit from `en.json` as the master translation and fallback for incomplete translations.
+You can get the correct tag and native name (has to start with capital letter) from the [list of ISO 639-1 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes). If necessary also add the corresponding [country code](https://en.wikipedia.org/wiki/ISO_3166-1#Officially_assigned_code_elements).
```

Comment:
Still haven't reached complete consensus on this, but I think we want to reference BCP47 here. https://tools.ietf.org/html/bcp47 From what I've seen this is the reference that browsers follow. Open to other arguments here, but at first glance, since we're choosing the default based on the browsers reported language, we should inherit the language tag standard from whatever browsers report.